source/whats-new.txt

```diff
@@ -32,6 +34,105 @@ Learn what's new in:
 
 .. _c-upcoming-breaking-changes:
 
+.. _c-version-1.29.0:
+
+What's New in 1.29.0
+--------------------
+
+The 1.29.0 release of ``libbson`` and ``libmongoc``, the libraries constituting 
+the C driver, includes several changes.
```

Comment:
S; this middle clause is kind of confusing to parse. Do we need to name these libraries or can we just refer to the C driver?

Maybe, you can include a note on this page that says something to the effect of "The driver release refers to the release of ... which are the libraries that constitute the driver

Reply:
I added a couple sentences to the overview section and then replaced this to just mention the driver. Also changed all instances to constants